### PR TITLE
(PDB-1571) update release notes for 2.3.5 release

### PR DESCRIPTION
**Pull request by wkalt**
this updates our release notes for 2.3.5

### CODE REVIEW

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -8,6 +8,57 @@ canonical: "/puppetdb/latest/release_notes.html"
...
+### Changes
+
+#### Bug Fixes and Maintenance
+
+* We have fixed a bug that caused PuppetDB to reject resource relationship
+  titles containing newlines. ([PDB-1529](https://tickets.puppetlabs.com/browse/PDB-1529))
+
+* We have removed an extraneous dependency on rubygem-json, which was
+  preventing the PuppetDB terminus from installing with the all in one agent on rhel 6.
+  ([PDB-1469](https://tickets.puppetlabs.com/browse/PDB-1469))
```
Comment (rbrw) [inline documentation/release_notes.markdown]: @wkalt Should rhel be RHEL?

File: documentation/release_notes.markdown [hunk 2]
```diff
@@ -8,6 +8,57 @@ canonical: "/puppetdb/latest/release_notes.html"
...
+
+* We have fixed a bug that caused unneccessary contention on the certnames
+  table during command submission, which in turn caused commands to be retried
+  more often than necessary.
+  ([PDB-1550](https://tickets.puppetlabs.com/browse/PDB-1550))
+
+* We have added additional error handling around PuppetDB startups that fail
+  due to implicit downgrades of ActiveMQ that can occur when users downgrade
+  PuppetDB. ([PDB-1454](https://tickets.puppetlabs.com/browse/PDB-1454))
```
Comment (rbrw) [inline documentation/release_notes.markdown]: @wkalt wondered about something like "...PuppetDB startup failures that  can be caused by server downgrades that implicitly downgrade ActiveMQ" instead.

**Comment by rbrw** [comment]
Overall looks good.

**Comment by wkalt** [comment]
@rbrw thanks -- updated

**Pull request merged**